keel-lemur/src/main/kotlin/com/netflix/spinnaker/keel/lemur/LemurCertificateResolver.kt

```diff
@@ -12,7 +12,7 @@ import org.springframework.stereotype.Component
 @Component
 @ConditionalOnBean(LemurService::class)
 class LemurCertificateResolver(
-  private val lemurService: LemurService
+  private val lemurCertificateByName: suspend (String) -> LemurCertificateResponse
```

Comment:
Inject the function into the resolver instead of the entire service

Reply:
This is pretty cool. First time I've seen injecting a function as a `@Bean`, but makes perfect sense.

Reply:
Yeah, I like this pattern. It's super friendly to tests as well.

Reply:
This is kind of how I'd like to approach refactoring the repositories if I ever get the chance. I'm curious to see whether it scales in a sane way if a component needs a bunch of such functions.